java/bulk-import/bulk-import-starter/src/main/java/sleeper/bulkimport/starter/executor/persistent/PersistentEmrPlatformExecutor.java

```diff
@@ -62,7 +71,16 @@ public void runJobOnPlatform(BulkImportArguments arguments) {
                 .build();
 
         LOGGER.info("Adding job flow step {}", addJobFlowStepsRequest);
-        emrClient.addJobFlowSteps(addJobFlowStepsRequest);
+        try {
+            emrClient.addJobFlowSteps(addJobFlowStepsRequest);
+        } catch (EmrException e) {
+            String message = e.awsErrorDetails().errorMessage();
+            if (Pattern.matches("Maximum number of active steps.+ for cluster exceeded\\.", message)) {
+                returnToQueueWhenClusterIsFull.send(arguments.getBulkImportJob());
```

Comment:
Add a log line here to report that it's being returned to the queue with a delay